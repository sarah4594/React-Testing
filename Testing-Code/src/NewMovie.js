import React, { Component } from 'react'
import MovieForm from './MovieForm'

class NewMovie extends Component {
  render() {
    return (
      <div>
        <h1 data-testid="page-title">New Movie</h1>
        <MovieForm />
      </div>
    )
  }
}

export default NewMovie
