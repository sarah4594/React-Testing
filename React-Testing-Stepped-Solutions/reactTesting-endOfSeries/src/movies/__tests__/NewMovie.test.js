import React from 'react';
import { render, cleanup } from 'react-testing-library';
import NewMovie from '../NewMovie';

afterEach(cleanup);

test('<NewMovie>', () => {
  const { getByTestId, queryByTestId, container } = render(<NewMovie />);
  expect(getByTestId('page-title').textContent).toBe('New Movie');
  expect(queryByTestId('movie-form')).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});
