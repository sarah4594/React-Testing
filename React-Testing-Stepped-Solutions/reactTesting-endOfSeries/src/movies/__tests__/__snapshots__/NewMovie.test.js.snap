// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie> 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    New Movie
  </h1>
  <form
    data-testid="movie-form"
  >
    <label
      for="text"
    >
      Text
      <input
        type="text"
        id="text"
      />
    </label>
    <button>
      Submit
    </button>
  </form>
</div>
`;
